### DISCUSSION TITLE
Creating custom CRUD routes

### DISCUSSION THREAD

**Discussion by MBafna19** [mockoon/mockoon#1384] [Help]
Hi all,

I'm working on emulating an existing API and I wanted to utilize data buckets. I have pre-defined POST requests that I'd like to insert data into a data bucket with. However, the syntax of these requests are different than CRUD route syntax. For example:

1. POST on my API - /insert vs POST on CRUD Routes /resources

It would be great if there was a way to set my API to do the same thing the CRUD Route API is doing. Is this possible today?

**Comment by 255kb** [comment]
Hi 
Indeed, this is not possible. 
While I understand how it could be valuable to you, I am not sure it's something we want to tackle in a near future.
The CRUD routes were designed to be a good alternative to tools like json-server and follow most REST principles. 
But we know REST is malleable and there are as many REST flavors as there are devs :) 

> Mockoon is an open-source project and relies on your support. If you found this answer helpful, please consider [sponsoring us](https://github.com/sponsors/mockoon). Additionally, we're in the process of developing a [SaaS platform](https://mockoon.com/pro/) to ensure the project's long-term sustainability. Currently, our platform offers an AI assistant, data synchronization, and team collaboration features, with cloud deployments coming soon. Thank you for your support!
